src/Engine/ProtoCore/DSASM/Executive.cs

```diff
@@ -817,6 +817,36 @@ private List<StackValue> PopArgumentsFromStack(int argumentCount)
             return sv;
         }
 
+        // cached values for repeated calls of the same object type and dispose function.
+        private int previousClassIndex;
+        private string previousProcedureName;
+        private CallSite callsite;
+
+        public StackValue CallDipose(ProcedureNode fNode,
+                                StackValue svThisPtr,
+                                int classIndex,
+                                ref bool explicitCall)
+        {
+            if (null != Properties.executingGraphNode)
+            {
+                exe.ExecutingGraphnode = Properties.executingGraphNode;
+            }
+
+            if (callsite == null || classIndex != previousClassIndex || previousProcedureName != fNode.Name)
+            {
+                previousClassIndex = classIndex;
+                previousProcedureName = fNode.Name;
+                callsite = new CallSite(classIndex, fNode.Name, exe.FunctionTable, runtimeCore.Options.ExecutionMode);
+            }
+
+            Validity.Assert(null != callsite);
+
+            StackValue sv = StackValue.Null;
+            sv = callsite.DispatchDispose(svThisPtr, runtimeCore);
```

Comment:
We can simply combine these statements into one:
```
StackValue sv = callsite.DispatchDispose(svThisPtr, runtimeCore);
```

Reply:
Good call... fixed